File: spiritdex-api/src/main/java/com/spiritdex/api/util/TypeColors.java
package com.spiritdex.api.util;

import java.util.Map;

/**
 * 18 属性 → 配色（hex）。前后端语义一致（前端 lib/type-colors.ts 同步）。
 * 用于属性徽章、相克矩阵染色。
 */
public final class TypeColors {

    private static final Map<String, String> BY_SLUG = Map.ofEntries(
            Map.entry("normal", "#9099A0"),
            Map.entry("grass", "#5DBE62"),
            Map.entry("fire", "#F0832E"),
            Map.entry("water", "#3B95F2"),
            Map.entry("light", "#F2C94C"),
            Map.entry("ground", "#B5793C"),
            Map.entry("ice", "#7FD4E8"),
            Map.entry("dragon", "#8A6FE8"),
            Map.entry("electric", "#F2C94C"),
            Map.entry("poison", "#9B59B6"),
            Map.entry("bug", "#A8B820"),
            Map.entry("fighting", "#C0392B"),
            Map.entry("flying", "#8AA9D6"),
            Map.entry("cute", "#EC6FA8"),
            Map.entry("ghost", "#6C5B7B"),
            Map.entry("dark", "#4A4A4A"),
            Map.entry("machine", "#95A5A6"),
            Map.entry("illusion", "#B370E8")
    );

    private TypeColors() {
    }

    /** 取属性配色，未知返回中性灰。 */
    public static String of(String slug) {
        return BY_SLUG.getOrDefault(slug, "#9099A0");
    }
}

File: spiritdex-web/lib/type-colors.ts
/**
 * 中文属性名 → 配色（与后端 TypeColors.java 语义一致）。
 * 用于无后端 type.color 时的兜底染色（卡片徽章 / 矩阵单元格）。
 */
const COLORS: Record<string, string> = {
  普通: "#9099A0",
  草: "#5DBE62",
  火: "#F0832E",
  水: "#3B95F2",
  光: "#F2C94C",
  地: "#B5793C",
  冰: "#7FD4E8",
  龙: "#8A6FE8",
  电: "#F2C94C",
  毒: "#9B59B6",
  虫: "#A8B820",
  武: "#C0392B",
  翼: "#8AA9D6",
  萌: "#EC6FA8",
  幽: "#6C5B7B",
  恶: "#4A4A4A",
  机械: "#95A5A6",
  幻: "#B370E8",
};

export function typeColor(name?: string | null): string {
  if (!name) return "#9099A0";
  return COLORS[name] ?? "#9099A0";
}

export function typeColorStyle(name?: string | null): React.CSSProperties {
  return { backgroundColor: typeColor(name), color: "#fff" };
}
